k8s/operators/src/diskpool/main.rs

```diff
@@ -479,11 +479,11 @@ impl ResourceContext {
 
                         // We expected the control plane to have a spec for this pool. It didn't so
                         // set the CRD to the error state and don't try to recreate it.
-                        self.mark_error().await
+                        self.mark_pool_not_found().await
                     }
                 } else if response.status() == clients::tower::StatusCode::SERVICE_UNAVAILABLE || response.status() == clients::tower::StatusCode::REQUEST_TIMEOUT {
                     // Probably grpc server is not yet up
-                    self.mark_unknown().await
+                    Ok(Action::requeue(Duration::from_secs(5)))
```

Comment:
I think so . Should we do it at first ServiceUnavailable itself? Thinking of marking it as None. Contaol plane status is Option anyway.

Reply:
Yes I think so, because we cannot determine what the status is, should be None.

Reply:
we dont know so maybe lets not post anything. These erros are connectivity/gRPC issues. So dont know what is the status .

Reply:
The problem is that if we don't change state, and the rest/grpc is not back up for whatever reason, we'll keep saying a pool is online for example, when it might not be.
I don't know what the common practice is in k8s for this, anyone knows? @Abhinandan-Purkait  @niladrih

Reply:
I was thinking of not showing it at all. marking as None. We dont know will not show

Reply:
Setting as None